exonum/src/node/mod.rs

```diff
@@ -909,6 +910,7 @@ impl Node {
     /// Creates node for the given services and node configuration.
     pub fn new(
         database: impl Into<Arc<dyn Database>>,
+        external_runtimes: impl IntoIterator<Item = impl Into<(u32, Box<dyn Runtime>)>>,
```

Comment:
Just out of curiosity. Is there really reason to use such complicated type rather then `Vec<(u32, Box<dyn Runtime)>` or something like this ?

Reply:
Well, I thought about it too. We have even a much funnier place: function takes `IntoIter` just to `collect` it into a `vec`, and then (!) call `into_iter` for it to pass into another function: [src](https://github.com/exonum/exonum/blob/dynamic_services/exonum/src/blockchain/builder.rs#L75).

However, I wanted this change to be consistent with surrounding code, so I made it `IntoIter` just as `services`.

I think it's a good topic for further refactoring, but fixing the whole issue is a bit out of scope of this PR.

Reply:
Ok. Obviously it makes sense for now.